### PR TITLE
Make use of the common adr template in the .github repo

### LINKED ISSUE #1415: celestiaorg ADR template
## Context

celestia-app made a small change to the ADR template in https://github.com/celestiaorg/celestia-app/pull/1407 so the ADR template in celestia-app differs from other celestiaorg repos.

## Proposal

Establish an org wide ADR template. If an org wide template is created, here are some proposed modifications:

- Include an authors section so they can be pinged in future references (proposed by @nashqueue)
- Remove the changelog section because git commit history exists (proposed by @rootulp) 
- Hoist the ADR status to the top of the document (proposed by @musalbas)

## Alternatives

Preserve the existing ADR templates (one per repo). If we want consistency across the org, re-apply changes to all ADR templates. If consistency isn't necessary, let individual teams decide what belongs in their ADR template.

## Notes
The canonical ADR template likely should not live in celestia-app and should instead live in a different celestiaorg repo. This issue exists in celestia-app because a suitable org wide repo could not be found (cc @MSevey).

### PR DESCRIPTION
**Pull request by MSevey**
## Overview

To unify the ADR structure we added a template in the .github repo. This PR adds a helpful command to generate new ADRs from that template. 

Example:     https://asciinema.org/a/56X6g1bA9NpWq0FxW8wDMRJ3g

Closes #1415 

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

**Review by liamsi — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -127,3 +127,12 @@ test-cover:
 	@echo "--> Generating coverage.txt"
 	@export VERSION=$(VERSION); bash -x scripts/test_cover.sh
 .PHONY: test-cover
+
+## adr-gen: Download the adr template from the celestiaorg/.github repo and
+#create a new adr file with the provided adr number and title.
+#
+# Usage: make adr-gen NUM=016 TITLE=my-adr-title
```
Comment (rootulp) [inline Makefile:134]: The `make help` command doesn't print this correctly:

```
...
  adr-gen        Download the adr template from the celestiaorg/.github repo and
```

So proposal to summarize this text into one line:
```suggestion
## adr-gen: Download the ADR template from the celestiaorg/.github repo. Ex. `make adr-gen NUM=016 TITLE=my-adr-title`
```

**Review by rootulp — APPROVED** [review]
[non-blocking] I noticed that `make adr-gen` without any arguments copy + pastes the template into to `docs/architecture/adr--.md` because the make command assumes those variables to be defined. It would be nice (although totally optional) if the filename was `docs/architecutre/adr-template.md` if the arguments weren't specified.

**Review by evan-forbes — DISMISSED** [review]
utACK +1 to Rootul's comments

**Review by evan-forbes — APPROVED** [review]
I approved some sensible comments while Matt is OOO so we can push this through

**Review by rootulp — APPROVED** [review]

**Pull request merged**